### ISSUE TITLE
Cannot start service at setup. Setup failed

### ISSUE THREAD

**Issue by juanmanuelrios** [Alachisoft/NCache#23]
I've downloaded latest version of ncache open source edition and is failing to complete installation process.
The issue is caused by service cannot be started during setup. Event Viewer display some errors related to NCacheSvc, such as:

No se puede iniciar el servicio. System.NullReferenceException: Referencia a objeto no establecida como instancia de un objeto.
   en Alachisoft.NCache.SocketServer.ServiceHost.Start(String[] args)
   en System.ServiceProcess.ServiceBase.ServiceQueuedMainCallback(Object state)

**Comment by faysalim** [comment]
It worked on my machine. The difference is that my system language is English and you seem to have Spanish. You can try changing the language or wait for NCache support to respond to this issue.

**Comment by Kal-Alachisoft** [comment]
Hi @juanmanuelrios ,

Please note that this is an already reported issue on our end, as mentioned by @faysalim it usually happens when the language is not English.

Whenever NCache gets installed on a machine it picks up on of the IPs present on the box and the NCache Services starts to listen on this IP. In this issue, the NCache is unable to get the IP from the OS and hence the value of IP is set to be null. Engineering is working on it to fix this, I will keep you updated regarding this.

Furthermore, a temporary work around for this is described in the steps below.

1.	Run the NCache installer as Administrator using the msiexec /I “installation path”.

2.	When the installer reaches the “Starting NCache service” part of the installation, please go to the directory “C:\Program Files\NCache\bin\service\Alachisoft.NCache.Service.exe.config” and open the NCache service configuration file.

3.	Look for the tags <add key="NCacheServer.BindToIP" value="xx.xx.xx.xx"/>and add your machine’s IP in the value part of both the tags. Save the file once the IP address has been added successfully.

4.	Hit the retry option from the dialogue box that you receive during the installation and the service should get started now.

Moreover, if you have NCache already installed you just update the value of the IP in the NCache Service config file as mentioned above and it should work just fine.

Please let us know if this worked for you.

**Comment by juanmanuelrios** [comment]
Hi @Kal-Alachisoft and @faysalim,

Many thanks for your response. 

I realized that my computer was refusing to start another services too, so I decided to make a fresh install of operating system. After that, installer works fine.

I couldn't get a good reason for the installer to fail, but it's clear now that some issue was interfering on services running on this computer.

Sorry for make a waste of time reviewing this issue. 

Best regards.
